### ISSUE TITLE
esbuild minifies CSS in a way that can break in old browsers

### ISSUE THREAD

**Issue by 19Qingfeng** [remix-run/remix#6114]
### What version of Remix are you using?

@remix-run/dev 1.15.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

I found this compilation of styles.

<img width="1510" alt="image" src="https://user-images.githubusercontent.com/31119562/233360489-20f427d9-023e-4b3e-bf77-75baf2ea161a.png">

when i use postcss-preset-env compile my style code, eg: `body { position: fixed; inset: 0; }`

First, it will be processed by postcss,get `body { position: fixed; top:0; bottom:0; left:0; right:0; } `. 

Then esbuild will process my style code,This will translate into `body { position: fixed; inset:0; }`

### Expected Behavior

Keep The postcss-preset-env dealing with me.

### Actual Behavior

Now, then esbuild will process my css code after then postcss.

**Comment by machour** [comment]
Hey @19Qingfeng, it would help a lot if you could provide a repository reproducing the problem.

**Comment by markdalgleish** [comment]
I had a look into this. It's actually not strictly related to PostCSS as you can manually write a CSS file with your example (`body { position: fixed; top:0; bottom:0; left:0; right:0; }`) and get the same result.

This is more of an issue with esbuild's CSS minification logic which is quite aggressive by default. I was able to fix it for CSS file imports by setting the `targets` array on the build inside `cssImportsPlugin`. I tried setting it to the list of browser versions needed for native JS module support since that's the absolute baseline for Remix:

```js
target: [
  "chrome61",
  "edge16",
  "safari11",
  "ios11",
  "firefox60",
  "opera48",
],
```

This worked for CSS file imports (i.e. `import href from "./styles.css"`) but this threw errors during our CSS bundle build since that also bundles JS, and esbuild can't handle certain JS features when targeting these browser versions.

I can fix this issue (at least for the default `remix` template) by raising the browser versions to the following:

```js
target: [
  "chrome63",
  "edge79",
  "safari12",
  "ios12",
  "firefox60",
  "opera50",
],
```

However, this feels a bit brittle to me since we can't decouple CSS and JS minification options and we might run into further issues with other JS features. I think the safer approach would be to look into a custom CSS minification step in our compiler (maybe using Lightning CSS) but this is a bigger job.

**Comment by markdalgleish** [comment]
Is this causing an actual user-facing issue for you in terms of browser support? Are there specific browsers you're trying to support that don't support CSS `inset`? https://caniuse.com/mdn-css_properties_inset

**Comment by 19Qingfeng** [comment]
> Is this causing an actual user-facing issue for you in terms of browser support? Are there specific browsers you're trying to support that don't support CSS `inset`? https://caniuse.com/mdn-css_properties_inset

Yes,I have some user browsers that don't support color abbreviations eg: #0009 not support on chrome 70.At present, my solution is use pnpm patch '@remix-run/dev' and add target in esbuild.

**Comment by 19Qingfeng** [comment]
> Is this causing an actual user-facing issue for you in terms of browser support? Are there specific browsers you're trying to support that don't support CSS `inset`? https://caniuse.com/mdn-css_properties_inset



> I had a look into this. It's not strictly related to PostCSS because you can manually write a CSS file containing your example code (`body { position: fixed; top:0; bottom:0; left:0; right:0; }`) and get the same result.
> 
> This is more of an issue with esbuild's CSS minification logic which is quite aggressive by default. I was able to fix it for CSS file imports by setting the `targets` array on the build inside `cssImportsPlugin`. I tried setting it to the list of browser versions needed for native JS module support since that's the absolute baseline for Remix:
> 
> ```js
> target: [
>   "chrome61",
>   "edge16",
>   "safari11",
>   "ios11",
>   "firefox60",
>   "opera48",
> ],
> ```
> 
> This worked for CSS file imports (i.e. `import href from "./styles.css"`) but this threw errors during our CSS bundle build since that also bundles JS, and esbuild can't handle certain JS features when targeting these browser versions.
> 
> I can fix this issue (at least for the default `remix` template) by raising the browser versions to the following:
> 
> ```js
> target: [
>   "chrome63",
>   "edge79",
>   "safari12",
>   "ios12",
>   "firefox60",
>   "opera50",
> ],
> ```
> 
> However, this feels a bit brittle to me since we can't decouple CSS and JS minification options and we might run into further issues with other JS features. I think the safer approach would be to look into a custom CSS minification step in our compiler (maybe using Lightning CSS) but this is a bigger job.

I always think exposing the browserlist is a good op, I saw this discussion, but I don't know what the remix team is choice right now.

**Comment by brophdawg11** [comment]
Unfortunately we had to revert the fix for this in https://github.com/remix-run/remix/pull/7324 as it caused some other issues with CSS features.  Going to re-open this so we can address it via browserslist support early in v2.

**Comment by brophdawg11** [comment]
Thank you for opening this issue, and our apologies we haven't gotten around to it yet!  

With the release of [React Router v7](https://remix.run/blog/react-router-v7) we are sun-setting continued development/maintenance on Remix v2.  If you have not already [upgraded to React Router v7](https://reactrouter.com/upgrading/remix), we recommend you do so. We've tried to make the upgrade process as smooth as possible with our [Future Flags](https://remix.run/docs/en/main/start/future-flags). We are now in the process of cleaning up outdated issues and pull requests to improve the overall hygiene of our repositories.  

We plan to continue to address 2 types of issues in Remix v2:

* Bugs that pose security concerns
* Bugs that prevent upgrading to React Router v7

If you believe this issue meets one of those criteria, please respond or [create a new issue](https://github.com/remix-run/remix/issues/new?template=bug_report.yml).

For all other issues, ongoing maintenance will be happening in React Router v7, so:

* If this is a bug, please reopen this issue [in that repo](https://github.com/remix-run/react-router/issues/new?template=bug_report.yml) with a new [minimal reproduction](https://reactrouter.com/new) against v7
* If this is a feature request, please open a new [Proposal Discussion](https://github.com/remix-run/react-router/discussions/new?category=proposals) in React Router, and if it gets enough community support it can be considered for implementation

If you have any questions you can always reach out on [Discord](https://rmx.as/discord). Thanks again for providing feedback and helping us make our framework even better!
